lib/patches/ancestry_patch.rb

```diff
@@ -8,6 +8,16 @@ def update_descendants_with_new_ancestry
       end
     end
   end
+
+  def parent=(parent)
+    super
+    clear_memoized_instance_variables
```

Comment:
would feel a little more comfortable returning the original value since people often type `x = y = z`

```ruby
  def parent=(parent)
    super.tap { clear_memoized_instance_variables }
  end
```

Reply:
Huh, I guess ruby had my back on this one...

```ruby
class Foo
  attr_reader :bar

  def bar= bar_val
    @bar = bar_val
    puts "IT WORKED!"
  end
end

puts (Foo.new.bar = "bar")
puts baz = Foo.new.bar = "baz"
puts "baz = #{baz}"

#=> IT WORKED!   <<<<< output from first bar=
#=> bar          <<<<< output from first inline puts
#=> IT WORKED!   <<<<< output from second bar=
#=> baz          <<<<< output from second inline puts
#=> baz = baz    <<<<< output from third inline puts
```

Apparently `=`'s methods don't behave the same as normal methods and always return the input?  TMYK 🌈✨🌠

Reply:
Anyway, adding a test because this could end up going bad if this is ever NOT the case.